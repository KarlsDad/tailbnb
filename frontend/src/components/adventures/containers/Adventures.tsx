import * as React from 'react';

// Function
import AdventureCard from '../functions/AdventureCard';

export const Adventures: React.FC<{}> = () => {
  return (
    <div className='w-full flex justify-center bg-gray-150'>
      <div className='w-full lg:max-w-6xl flex justify-start items-start flex-wrap py-32'>
        <div
          style={{ fontFamily: 'airbnb-bold' }}
          className='w-full md:w-4/12 h-full text-5xl text-gray-750 leading-none flex items-center'>
          <h3>Airbnb Adventures</h3>
        </div>
        <div
          style={{ fontFamily: 'airbnb-book' }}
          className='w-full md:w-8/12 px-4 text-gray-750 flex'>
          <AdventureCard
            icon='camp'
            title='Meals and stays included'
            description='Every adventure is fully planned, from activities to accommodations.'
          />
          <AdventureCard
            icon='map'
            title='Led by locals'
            description='Adventures are hosted by local insiders who immerse you in their culture.'
          />
          <AdventureCard
            icon='campcar'
            title='One-of-a-kind access'
            description="Explore places and communities you can\'t easily access on your own."
          />
        </div>
      </div>
    </div>
  );
};
