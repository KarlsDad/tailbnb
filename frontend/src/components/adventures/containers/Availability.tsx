import * as React from 'react';

// Functions
import AvailabilityCard from '../functions/AvailabilityCard';

export const Availability: React.FC<{}> = () => {
  return (
    <div className='w-full flex justify-center'>
      <div className='px-4 md:px-16 w-full max-w-6xl xl:px-0 flex justify-start items-start flex-wrap py-16'>
        <div
          style={{ fontFamily: 'airbnb-bold' }}
          className='text-2xl w-full md:text-3xl text-gray-750 mb-8'>
          Availability
        </div>
        <div className='w-full overflow-y-hidden'>
          <div className='w-full h-full overflow-y-hidden'>
            <div className='w-full h-full scroller'>
              <div className='w-full scrollable sm:inset-x-0 flex items-start justify-start py-2 rounded-xl md:w-full'>
                <div className='mr-1'>
                  <AvailabilityCard />
                </div>
                <div className='mr-1'>
                  <AvailabilityCard />
                </div>
              </div>
            </div>
          </div>
        </div>
        <button className='border-2 border-green-850 rounded mt-24'>
          <div className='py-2 px-4'>
            <p style={{ fontFamily: 'airbnb-bold' }} className='text-green-850'>
              Show more availability
            </p>
          </div>
        </button>
      </div>
    </div>
  );
};
