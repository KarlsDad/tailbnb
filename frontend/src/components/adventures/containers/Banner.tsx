import * as React from 'react';

// Function
import { BannerCard } from '../functions/BannerCard';

// Image

export const Banner: React.FC<{}> = () => {
  return (
    <div className='w-full bg-black flex justify-center'>
      <div className='px-4 md:px-16 w-full max-w-6xl xl:px-0 py-12'>
        <div className='w-full flex'>
          <div className='w-2/7 border-r-4 border-black relative bg-cover bg-center overflow-hidden'>
            <img
              className='h-full w-full'
              src='https://a0.muscache.com/im/pictures/a5845dc3-6d54-4864-a383-4dffbc16c20e.jpg?aki_policy=exp_md'
              alt=''
            />
          </div>
          <div className='w-2/7 border-r-4 border-black relative overflow-hidden'>
            <img
              className='h-full object-cover'
              src='https://a0.muscache.com/im/pictures/2cde86ed-9fa5-4776-952a-c83e0afda996.jpg?aki_policy=exp_md'
              alt=''
            />
          </div>
          <div className='border-black w-1/7 relative overflow-hidden'>
            <div className='w-full bg-blue-500 h-1/2'>
              <img
                className='h-full object-cover'
                src='https://a0.muscache.com/im/pictures/lombard/MtTemplate-1317456-media_library/original/45cba600-ef28-4667-823e-453c31944b45.jpeg?aki_policy=exp_md'
                alt=''
              />
            </div>
            <div className='w-full bg-blue-300 h-1/2'>
              <img
                className='h-full object-cover'
                src='https://a0.muscache.com/im/pictures/lombard/MtTemplate-1317456-media_library/original/5d0cbc89-ed65-4063-9ec2-83eb7690bbd7.jpeg?aki_policy=exp_md'
                alt=''
              />
            </div>
          </div>
          <div className='border-l-4 border-black w-2/7 relative overflow-hidden'>
            <img
              className='h-full object-cover'
              src='https://a0.muscache.com/im/pictures/4177e9be-7c5e-4c05-9b79-f4afd144a389.jpg?aki_policy=exp_md'
              alt=''
            />
          </div>
        </div>
        <div className='w-full lg:max-w-6xl flex justify-start items-start flex-wrap'>
          <div className='w-full md:w-1/2 lg:w-4/12'>
            <div
              style={{ fontFamily: 'airbnb-bold' }}
              className='items-end justify-start pt-6 text-xs uppercase text-white mb-1'>
              2-Day Adventure
            </div>

            <p
              style={{ fontFamily: 'airbnb-book' }}
              className='w-full text-3xl text-white leading-tight'>
              Atlas Mountains Experience
            </p>
            <p
              style={{ fontFamily: 'airbnb-book' }}
              className='w-full text-white '>
              Marrakech, Morocco
            </p>
            <div className='flex justify-start text-white my-2'>
              <p style={{ fontFamily: 'airbnb-bold' }}>5.0</p>
              <div className='h-3 w-3 mx-1'>
                <svg
                  className='w-full h-full'
                  style={{ fill: '#ffffff', marginTop: 6 }}
                  viewBox='0 0 511.99143 511'
                  xmlns='http://www.w3.org/2000/svg'>
                  <path d='m510.652344 185.882812c-3.371094-10.367187-12.566406-17.707031-23.402344-18.6875l-147.796875-13.417968-58.410156-136.75c-4.3125-10.046875-14.125-16.53125-25.046875-16.53125s-20.738282 6.484375-25.023438 16.53125l-58.410156 136.75-147.820312 13.417968c-10.835938 1-20.011719 8.339844-23.402344 18.6875-3.371094 10.367188-.257813 21.738282 7.9375 28.925782l111.722656 97.964844-32.941406 145.085937c-2.410156 10.667969 1.730468 21.699219 10.582031 28.097656 4.757813 3.457031 10.347656 5.183594 15.957031 5.183594 4.820313 0 9.644532-1.28125 13.953125-3.859375l127.445313-76.203125 127.421875 76.203125c9.347656 5.585938 21.101562 5.074219 29.933593-1.324219 8.851563-6.398437 12.992188-17.429687 10.582032-28.097656l-32.941406-145.085937 111.722656-97.964844c8.191406-7.1875 11.308594-18.535156 7.9375-28.925782zm-252.203125 223.722657' />
                </svg>
              </div>
              <p>(1)</p>
            </div>
            <div className='flex justify-start'>
              <div
                style={{ fontFamily: 'airbnb-medium' }}
                className='text-2xs mt-2 mr-2 mb-2 py-1 px-2 bg-gray-950 text-white uppercase rounded'>
                Hiking
              </div>
              <div
                style={{ fontFamily: 'airbnb-medium' }}
                className='text-2xs mt-2 mr-2 mb-2 py-1 px-2 bg-gray-950 text-white uppercase rounded'>
                Trekking
              </div>
            </div>
          </div>
          <div className='w-full md:w-1/2 lg:w-8/12 h-full'>
            <div className='w-9/12 mt-4'>
              <p
                style={{ fontFamily: 'airbnb-book' }}
                className='py-4 text-white md:text-lg md:py-10'>
                Trek & Cook with the berbers
              </p>
            </div>
          </div>
          <div className='w-full lg:w-1/2  lg:w-8/12'>
            <div className='w-full flex justify-start flex-wrap'>
              <div className='flex w-1/2'>
                <BannerCard type='Duration' value={2} />
                <BannerCard type='Group size' value={6} />
              </div>
              <div className='flex w-1/2'>
                <BannerCard type='Activity level' value='moderate' />
                <BannerCard type='Hosted in' value='English' />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
