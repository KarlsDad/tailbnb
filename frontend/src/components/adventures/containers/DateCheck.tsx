import * as React from 'react';

export const DateCheck: React.FC<{}> = () => {
  return (
    <div className='bg-white w-screen py-4 sticky bottom-0 z-50 absolute flex justify-between'>
      <div className='flex'>
        <div>img here</div>
        <div className='flex'>
          <p>Sailing Adventures in Howe Sound Fjord</p>
          <div className='flex'>
            <p>4.97</p>
            <div>svg here</div>
          </div>
          <p>(233)</p>
        </div>
      </div>
      <div className='flex'>
        <p>$119/person</p>
        <button>
          <div>
            <p>See dates</p>
          </div>
        </button>
      </div>
    </div>
  );
};
