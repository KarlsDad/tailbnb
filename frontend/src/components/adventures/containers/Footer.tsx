import * as React from 'react';

export const Footer: React.FC<{}> = () => {
  return <div className='w-screen border-t border-gray-300'>Hello</div>;
};
