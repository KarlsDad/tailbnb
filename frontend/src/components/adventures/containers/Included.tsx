import * as React from 'react';

// Function
import IncludedCard from '../functions/IncludedCard';

export const Included: React.FC<{}> = () => {
  return (
    <div className='w-full flex justify-center'>
      <div className='px-4 md:px-16 w-full max-w-6xl xl:px-0 flex justify-start items-start flex-wrap py-16'>
        <div
          style={{ fontFamily: 'airbnb-bold' }}
          className='w-full md:w-4/12 md:h-full text-2xl md:text-3xl text-gray-750 pb-8 md:pb-0'>
          What's included
        </div>
        <div
          style={{ fontFamily: 'airbnb-book' }}
          className='w-full md:w-8/12 md:px-4 text-gray-750 flex overflow-x-hidden overflow-y-hidden md:overflow-x-visible md:overflow-y-vidi
          '>
          <div className='overflow-y-hidden'>
            <div className='w-full h-full overflow-y-hidden'>
              <div className='h-full scroller'>
                <div className='scrollable sm:inset-x-0 flex items-start justify-start py-2 rounded-xl w-104 w-full md:w-full'>
                  <div className='mr-4'>
                    <IncludedCard />
                  </div>
                  <div className='mr-4'>
                    <IncludedCard />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
