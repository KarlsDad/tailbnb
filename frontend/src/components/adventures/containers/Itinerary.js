import React from 'react';

// Images
const detail2 = require('../../../../public/img/high/staydetail2.jpg');

export default () => {
  return (
    <div className='w-full flex justify-center'>
      <div className='w-full lg:max-w-6xl flex justify-start items-start flex-wrap py-16'>
        <div
          style={{ fontFamily: 'airbnb-bold' }}
          className='w-full md:w-4/12 h-full text-3xl text-gray-750'>
          Your itinerary
        </div>
        <div className='w-full md:w-8/12'>
          <div
            style={{ fontFamily: 'airbnb-book' }}
            className='px-5 text-gray-750 border-l border-gray-800 relative '>
            <div
              style={{ height: 7, width: 7, top: -7, left: -3.6 }}
              className='absolute'>
              <svg
                className='h-full w-full'
                xmlns='http://www.w3.org/2000/svg'
                viewBox='0 0 512 512'>
                <g>
                  <g>
                    <path
                      d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
                    />
                  </g>
                </g>
              </svg>
            </div>
            <p className='mb-2'>Day 1</p>
            <div className='w-full flex justify-start items-start pb-8'>
              <div className='w-5/12'>
                <img className='h-full w-full' src={detail2} alt='' />
              </div>

              <div className='w-7/12 pl-5'>
                <p
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-base'>
                  Marrakech-Oughla - Tinzerte
                </p>
                <p className='text-base'>
                  Marrakech to imi oughlad small village situated 1,450m at the
                  base of Toubkal. Here we begin the gentle accent.
                </p>
                <button
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-green-850 text-sm mt-4'>
                  Show more
                </button>
              </div>
            </div>
            <div
              style={{ height: 7, width: 7, bottom: -7, left: -3.6 }}
              className='absolute z-100'>
              <svg
                className='h-full w-full'
                xmlns='http://www.w3.org/2000/svg'
                viewBox='0 0 512 512'>
                <g>
                  <g>
                    <path
                      d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
                    />
                  </g>
                </g>
              </svg>
            </div>
          </div>
          <div
            style={{ fontFamily: 'airbnb-book' }}
            className='relative px-5 text-gray-750'>
            <p className='mb-2'>Day 2</p>
            <div className='w-full flex justify-start items-start pb-8'>
              <div className='w-5/12'>
                <img className='h-full w-full' src={detail2} alt='' />
              </div>

              <div className='w-7/12 pl-5'>
                <p
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-base'>
                  Tizerte - Ouirgane - Marrakech
                </p>
                <p className='text-base'>
                  The day's trek will take you through the villages of Imariren
                  Noufla and Anraz. A picnic style lunch will be enjoyed..
                </p>
                <button
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-green-850 text-sm mt-4'>
                  Show more
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
