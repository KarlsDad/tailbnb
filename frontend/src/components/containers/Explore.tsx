import * as React from 'react';

// Components
import { ExploreCard } from '../presentational/ExploreCard';

// Images
// High Resolution
const explore1 = require('../../../public/img/high/explore1.jpg');
const explore2 = require('../../../public/img/high/explore2.jpg');
const explore3 = require('../../../public/img/high/explore3.jpg');

interface Explore {
  img: string;
  title: string;
}

const explores: Explore[] = [
  {
    img: explore1,
    title: 'Stay'
  },
  {
    img: explore2,
    title: 'Experience'
  },
  {
    img: explore3,
    title: 'Adventure'
  }
];

export const Explore: React.FC<{}> = () => {
  return (
    <div className='overflow-y-hidden'>
      <div className='w-full h-full overflow-y-hidden'>
        <div className='h-full scroller'>
          <div className='scrollable sm:inset-x-0 flex items-center justify-start py-2 rounded-xl w-80 md:w-full'>
            {explores.map(({ img, title }, index) => {
              return (
                <div className='w-32 mr-4 sm:w-1/3 cursor-pointer'>
                  <ExploreCard key={index} img={img} title={title} />
                </div>
              );
            })}
          </div>
        </div>
      </div>
    </div>
  );
};
