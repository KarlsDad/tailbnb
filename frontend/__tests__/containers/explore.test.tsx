import * as React from 'react';
import { shallow } from 'enzyme';

// Component
import { Explore } from '../../src/components/containers/Explore';

describe('Explore', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Explore />);
    expect(component).toMatchSnapshot();
  });
});
