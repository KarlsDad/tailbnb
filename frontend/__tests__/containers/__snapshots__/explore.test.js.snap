// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
